src/primitive/general_matrix_gate.jl

```diff
@@ -43,6 +43,8 @@ Create a [`GeneralMatrixBlock`](@ref) with a matrix `m`.
 """
```

Comment:
I would suggest to add a one line doc here to describe this behaviour.

Reply:
done

Reply:
It would be better to have it on `matblock` as well I think, which will warn the users when they construct. I'll do it...

Reply:
why... the inconsistancy happens only when you call `mat`

Reply:
because usually you only check the doc string of `matblock` rather than `mat` then use `mat` intuitively. I think it won't hurt to have some warnings in the docstring.

Reply:
Then, this docstring should state clearly, "when `mat` is called".